### PR TITLE
Added `/target/generated-sources/apt` to resources for maven and m2e

### PR DESCRIPTION
**Pull request by vigier**
Additionally it was needed to use the `build-helper-maven-plugin`, as
m2e would otherwise exclude by default all sources from the additional
source folder (s. https://stackoverflow.com/a/7759911 )

Signed-off-by: Peter Vigier <Peter.Vigier@bosch-si.com>

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

**Comment by StefanKlt** [comment]
👍 looks good, thanks

File: hawkbit-repository/hawkbit-repository-jpa/pom.xml [hunk 1]
```diff
@@ -91,6 +91,24 @@
    </dependencies>
 
    <build>
+      <resources>
+         <!-- We need to add also the default-resources, as this configuration overwrites the defaults -->
+         <resource>
+            <directory>src/main/java</directory>
```
Comment (StefanKlt) [inline hawkbit-repository/hawkbit-repository-jpa/pom.xml]: This default-resources seems to be included even if it is removed from this list. Please check if it is needed here.

File: hawkbit-repository/hawkbit-repository-jpa/pom.xml [hunk 2]
```diff
@@ -91,6 +91,24 @@
...
+         </resource>
+         <resource>
+            <directory>src/main/resources</directory>
+         </resource>
+         <resource>
+            <directory>target/generated-sources/apt</directory>
+         </resource>
+         <resource>
+            <directory>src/test/java</directory>
```
Comment (StefanKlt) [inline hawkbit-repository/hawkbit-repository-jpa/pom.xml]: This default-resources seems to be included even if it is removed from this list. Please check if it is needed here.

Comment (vigier) [reply hawkbit-repository/hawkbit-repository-jpa/pom.xml]: I removed the resource section completely as for adding an additional "source" folder the build-helper-plugin alone is enough.

**Pull request merged**